From stable+bounces-192344-greg=kroah.com@vger.kernel.org Tue Nov  4 10:33:06 2025
From: "jingxian.li" <jingxian.li@shopee.com>
Date: Tue,  4 Nov 2025 17:27:42 +0800
Subject: Revert "perf dso: Add missed dso__put to dso__load_kcore"
To: stable@vger.kernel.org, peterz@infradead.org, mingo@redhat.com, acme@kernel.org, mark.rutland@arm.com, alexander.shishkin@linux.intel.com, jolsa@kernel.org, namhyung@kernel.org, adrian.hunter@intel.com, sashal@kernel.org, linux-perf-users@vger.kernel.org, linux-kernel@vger.kernel.org
Cc: "jingxian.li" <jingxian.li@shopee.com>
Message-ID: <20251104092740.108928-3-jingxian.li@shopee.com>

From: "jingxian.li" <jingxian.li@shopee.com>

This reverts commit e5de9ea7796e79f3cd082624f788cc3442bff2a8.

The patch introduced `map__zput(new_node->map)` in the kcore load
path, causing a segmentation fault when running `perf c2c report`.

The issue arises because `maps__merge_in` directly modifies and
inserts the caller's `new_map`, causing it to be freed prematurely
while still referenced by kmaps.

Later branchs (6.12, 6.15, 6.16) are not affected because they use
a different merge approach with a lazily sorted array, which avoids
modifying the original `new_map`.

Fixes: e5de9ea7796e ("perf dso: Add missed dso__put to dso__load_kcore")

Signed-off-by: jingxian.li <jingxian.li@shopee.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 tools/perf/util/symbol.c |    1 -
 1 file changed, 1 deletion(-)

--- a/tools/perf/util/symbol.c
+++ b/tools/perf/util/symbol.c
@@ -1366,7 +1366,6 @@ static int dso__load_kcore(struct dso *d
 				goto out_err;
 			}
 		}
-		map__zput(new_node->map);
 		free(new_node);
 	}
 
